src/Stack/Setup.hs

```diff
@@ -1619,42 +1620,52 @@ chattyDownload label downloadInfo path = do
             , drLengthCheck = mtotalSize
             , drRetryPolicy = drRetryPolicyDefault
             }
-    x <- verifiedDownload dReq path chattyDownloadProgress
+    x <- verifiedDownload dReq path (chattyDownloadProgress label mtotalSize)
```

Comment:
what about adding it as something like 
```
verifiedDownloadWithProgress req path lbl msize = verifiedDownload req path (chattyDownloadProgress lbl msize)
```
into `Network.HTTP.StackClient` with `chattyDownloadProgress` as a part of it. That way we e.g. won't need o export `chattyDownloadProgress` from `Stack.Setup` which looks a bit odd

Reply:
@qrilka Thanks for the review! I moved some of the relevant code from `Setup` to `StackClient` module so hopefully all is good now.